sklearn/partial_dependence.py

```diff
@@ -0,0 +1,577 @@
+        averaged_predictions = averaged_predictions.reshape(1, -1)
+    elif is_classifier(est) and averaged_predictions.shape[0] == 2:
+        # Binary classification, shape is (2, n_points).
+        # we output the effect of **positive** class
+        averaged_predictions = averaged_predictions[1]
+        averaged_predictions = averaged_predictions.reshape(1, -1)
+
+    return averaged_predictions
+
+
+def partial_dependence(est, target_variables, X, percentiles=(0.05, 0.95),
+                       grid_resolution=100, method='auto'):
+    """Partial dependence of ``target_variables``.
+
+    .. _warning_recursion_init:
+
+    .. warning::
+        The 'recursion' method only works for gradient boosting estimators,
+        and unlike the 'brute' method, it does not account for the ``init``
+        predictor of the boosting process. In practice this will produce the
+        same values as 'brute' up to a constant offset in the target
+        response, provided that ``init`` is a consant estimator (which is
+        the default). However, as soon as ``init`` is not a constant
+        estimator, the partial dependence values are incorrect.
+
+    Read more in the :ref:`User Guide <partial_dependence>`.
+
+    Parameters
+    ----------
+    est : BaseEstimator
+        A fitted classification or regression model. Multioutput-multiclass
+        classifiers are not supported.
+    target_variables : list or array-like of int
+        The target features for which the partial dependency should be
+        computed.
+    X : array-like, shape=(n_samples, n_features)
+        ``X`` is used both to generate a grid of values for the
+        ``target_variables``, and to compute the averaged predictions when
+        method is 'brute'.
+    percentiles : tuple of float, optional (default=(0.05, 0.95))
+        The lower and upper percentile used to create the extreme values
+        for the ``grid``.
+    grid_resolution : int, optional (default=100)
+        The number of equally spaced points on the grid, for each target
+        feature.
+    method : str, optional (default='auto')
+        The method used to calculate the averaged predictions:
+
+        - 'recursion' is only supported for objects inheriting from
+          `BaseGradientBoosting`, but is more efficient in terms of speed.
+          With this method, ``X`` is only used to build the
+          grid. This method does not account for the ``init`` predicor of
+          the boosting process, which may lead to incorrect values (see
+          :ref:`this warning<warning_recursion_init>`).
+
+        - 'brute' is supported for any estimator, but is more
+          computationally intensive.
+
+        - If 'auto', then 'recursion' will be used for
+          ``BaseGradientBoosting`` estimators, and 'brute' used for other
+          estimators.
+
+    Returns
+    -------
+    averaged_predictions : array, shape=(n_targets, n_points)
```

Comment:
shouldn't we reshape this to `(n_targets, n_values0, n_values1, ...)`? Wouldn't that be more convenient (this reshape is done in `plot_partial_dependence` anyway), particularly when having to explain the correspondence to `values`?